### PR TITLE
Add more python tests

### PR DESCRIPTION
**Pull request by thebevrishot**
## PR intention
Add tests from the upper version of bitcoin for functionality that we lack tests.

## Code changes brief
- Add wallet encryption test (https://github.com/bitcoin/bitcoin/pull/10551)
- Add notification tests (https://github.com/bitcoin/bitcoin/pull/10941)
- Fix wallet functionality test and add more case(https://github.com/bitcoin/bitcoin/pull/15378)
- Add a lock to `SyncWithChain` to avoid hanging when try to encrypt wallet immediately after startup.

### CODE REVIEW

**Review by a-bezrukov — APPROVED** [review]

File: src/hdmint/wallet.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ std::pair<uint256,uint256> CHDMintWallet::RegenerateMintPoolEntry(CWalletDB& wal
 /**
  * Generate the mintpool for the current master seed.
  *
- * only runs if the current mintpool is exhausted and we need new mints (ie. the next mint to
```
Comment (psolstice) [inline src/hdmint/wallet.cpp:132]: Fixing formatting in unrelated files is evil

**Review by psolstice — APPROVED** [review]
Ok. Just no more changes in formatting please. Especially in untouched files
